universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java

```diff
@@ -191,4 +195,36 @@ public int hashCode() {
     }
     return pathsRemoved.build();
   }
+
+  public void setUserName(String displayName) {
+    String firstName;
+    String lastName = null;
+    String middleName = null;
+    List<String> listSplit = Splitter.on(' ').splitToList(displayName);
+    switch (listSplit.size()) {
+      case 2:
+        firstName = listSplit.get(0);
+        lastName = listSplit.get(1);
+        break;
+      case 3:
+        firstName = listSplit.get(0);
+        middleName = listSplit.get(1);
+        lastName = listSplit.get(2);
+        break;
+      case 1:
+        // fallthrough
+      default:
+        // Too many names - put them all in first name.
+        firstName = displayName;
+    }
+    setUserName(firstName, middleName, lastName);
+  }
+
+  private void setUserName(
+      String firstName, @Nullable String middleName, @Nullable String lastName) {
+    // We don't have a question for this yet, so this is unimplemented right now.
```

Comment:
question: If we're setting this on login, should we save it to the db? i.e. create a field on Applicant

Reply:
I don't really think so - I think it's more appropriate to have it be stored in applicant data once we have a "what is your name" question hardcoded in there.  You know more about how we're planning to store answers than me, though - is my theory about storing it in applicant data plausible?

Reply:
I think based on our discussion at standup Friday this is possible. Happy to go with this for now!